cdap-explore/src/main/java/co/cask/cdap/hive/stream/HiveStreamInputFormat.java

```diff
@@ -94,12 +106,99 @@ public InputSplit createSplit(Path path, Path indexPath, long startTime, long en
     Location streamPath = StreamUtils.createGenerationLocation(streamConfig.getLocation(),
                                                                StreamUtils.getGeneration(streamConfig));
 
-    // TODO: examine query where clause to get a better start and end time based on timestamp filters.
+    StreamInputSplitFinder.Builder builder = StreamInputSplitFinder.builder(streamPath.toURI());
+    return setupBuilder(conf, streamConfig, builder).build(SPLIT_FACTORY);
+  }
+
+  /**
+   * Setups the given {@link StreamInputSplitFinder.Builder} by analyzing the query.
+   */
+  private StreamInputSplitFinder.Builder setupBuilder(Configuration conf, StreamConfig streamConfig,
+                                                      StreamInputSplitFinder.Builder builder) {
     // the conf contains a 'hive.io.filter.expr.serialized' key which contains the serialized form of ExprNodeDesc
-    URI path = streamPath.toURI();
+    String serializedExpr = conf.get(TableScanDesc.FILTER_EXPR_CONF_STR);
+    if (serializedExpr == null) {
+      return builder;
+    }
+
+    // Analyze the query to extract predicates that can be used for indexing (i.e. setting start/end time)
+    IndexPredicateAnalyzer analyzer = new IndexPredicateAnalyzer();
+    for (CompareOp op : CompareOp.values()) {
+      analyzer.addComparisonOp(op.getOpClassName());
+    }
+
+    // Stream can only be indexed by timestamp
+    analyzer.clearAllowedColumnNames();
+    analyzer.allowColumnName("ts");
+
+    ExprNodeGenericFuncDesc expr = Utilities.deserializeExpression(serializedExpr);
+
+    List<IndexSearchCondition> conditions = Lists.newArrayList();
+    analyzer.analyzePredicate(expr, conditions);
```

Comment:
It doesn't handle "or" case as far as I tested. We can enhance the support later in this class.